File: 1512-number-of-good-pairs/1512-number-of-good-pairs.java
class Solution {
    public int numIdenticalPairs(int[] nums) {
        int arr[] = new int[101];
        for(int i: nums){
            arr[i]++;
        }
        int cnt=0;
        for(int i=0 ; i<101 ; i++)
        {
            cnt += arr[i]*(arr[i]-1)/2;
        }
        return cnt;
    }
}

File: 1512-number-of-good-pairs/1512-number-of-good-pairs.cpp
class Solution {
public:
    int fun(int n){
        return n*(n-1)/2;
    }
    int numIdenticalPairs(vector<int>& nums) {
        int res=0;
        map<int, int> mp;
        for(int i=0 ; i<nums.size() ; i++){
            mp[nums[i]]++;
        }
        for(auto it: mp){
            res += fun(it.second);
        }
        return res;
    }
};